cogs/dank_memes.py

```diff
@@ -63,6 +71,89 @@ class DankMemes(BaseCog):
+                  for i in image.shape]
+        out[tuple(coords)] = 1
+
+        # Pepper mode
+        num_pepper = numpy.ceil(amount * image.size * (1. - s_vs_p))
+        coords = [numpy.random.randint(0, i - 1, int(num_pepper))
+                  for i in image.shape]
+        out[tuple(coords)] = 0
+        return out
+
+    def _saturate(self, img):
+        """
+        frys an image
+
+        :param img: PIL.Image
+        :return PIL.Image:
+        """
+        flare_positions = []
+        opencv_img = cv2.cvtColor(numpy.array(img), cv2.COLOR_RGB2GRAY)
+        faces = cv2.CascadeClassifier(cv2.data.haarcascades + 'haarcascade_frontalface_default.xml')\
+            .detectMultiScale(opencv_img,
+                              scaleFactor=1.3,
+                              minNeighbors=5,
+                              minSize=(30, 30),
+                              flags=cv2.CASCADE_SCALE_IMAGE
+                              )
+        for (x, y, w, h) in faces:
+            face_roi = opencv_img[y:y+h, x:x+w]  # Get region of interest (detected face)
+            eyes = cv2.CascadeClassifier(cv2.data.haarcascades + 'haarcascade_eye.xml').detectMultiScale(face_roi)
+            for (ex, ey, ew, eh) in eyes:
+                eye_corner = (ex + ew / 2, ey + eh / 2)
+                flare_size = eh if eh > ew else ew
+                flare_size *= 4
+                corners = [math.floor(x) for x in eye_corner]
+                flare_positions.append(FlarePosition(*corners, flare_size))
+
+        # Crush image to hell and back
+        width, height = img.width, img.height
+        img = img.resize((int(width ** .75), int(height ** .75)), resample=Image.LANCZOS)
+        img = img.resize((int(width ** .88), int(height ** .88)), resample=Image.BILINEAR)
+        img = img.resize((int(width ** .9), int(height ** .9)), resample=Image.BICUBIC)
+        img = img.resize((width, height), resample=Image.BICUBIC)
+        img = ImageOps.posterize(img, 4)
+
+        # Generate colour overlay
+        r = img.split()[0]
+        r = ImageEnhance.Contrast(r).enhance(2.0)
+        r = ImageEnhance.Brightness(r).enhance(1.5)
+
+        r = ImageOps.colorize(r, (254, 0, 2), (255, 255, 15))
+
+        # Overlay red and yellow onto main image and sharpen the hell out of it
+        img = Image.blend(img, r, 0.75)
+        img = ImageEnhance.Sharpness(img).enhance(100.0)
+
+        # Apply flares on any detected eyes
+        if flare_positions:
+            flare_img = Image.open(STATIC_PATH + 'images/flare.png')
+            for flare in flare_positions:
+                flare_transformed = flare_img.copy().resize((flare.size,) * 2, resample=Image.BILINEAR)
+                img.paste(flare_transformed, (flare.x, flare.y), flare_transformed)
+
+        return img
+
+    def _deep_fry(self, image):
+        file = NamedTemporaryFile(suffix='.jpg')
+        Image.fromarray(self._salt_and_pepper(numpy.asarray(self._saturate(image.convert('RGB'))))
```

Comment:
Can you break this out into multiple lines so it's more clear what all is going on? This is a very busy line that does a ton of stuff to the image but it's hard to decipher like this